  0%|          | 0/250 [00:00<?, ?it/s]  0%|          | 1/250 [00:00<01:25,  2.90it/s]  1%|          | 2/250 [00:08<19:30,  4.72s/it]  1%|          | 3/250 [00:11<15:58,  3.88s/it]  2%|▏         | 4/250 [00:13<13:56,  3.40s/it]  2%|▏         | 5/250 [00:16<13:23,  3.28s/it]  2%|▏         | 6/250 [00:20<13:26,  3.31s/it]  3%|▎         | 7/250 [00:22<12:45,  3.15s/it]  3%|▎         | 8/250 [00:25<11:44,  2.91s/it]  4%|▎         | 9/250 [00:27<10:54,  2.71s/it]  4%|▍         | 10/250 [00:30<11:11,  2.80s/it]  4%|▍         | 11/250 [00:32<10:23,  2.61s/it]  5%|▍         | 12/250 [00:35<10:44,  2.71s/it]  5%|▌         | 13/250 [00:38<10:45,  2.72s/it]  6%|▌         | 14/250 [00:41<11:34,  2.94s/it]  6%|▌         | 15/250 [00:45<11:48,  3.01s/it]  6%|▋         | 16/250 [00:48<11:50,  3.04s/it]  7%|▋         | 17/250 [00:50<11:04,  2.85s/it]  7%|▋         | 18/250 [00:53<10:48,  2.80s/it]  8%|▊         | 19/250 [00:56<11:37,  3.02s/it]  8%|▊         | 20/250 [01:00<12:08,  3.17s/it]  8%|▊         | 21/250 [01:03<11:39,  3.05s/it]  9%|▉         | 22/250 [01:06<12:14,  3.22s/it]  9%|▉         | 23/250 [01:09<11:44,  3.10s/it] 10%|▉         | 24/250 [01:13<12:17,  3.26s/it] 10%|█         | 25/250 [01:17<13:15,  3.54s/it] 10%|█         | 26/250 [01:20<12:38,  3.39s/it] 11%|█         | 27/250 [01:23<12:07,  3.26s/it] 11%|█         | 28/250 [01:26<12:13,  3.30s/it] 12%|█▏        | 29/250 [01:28<10:56,  2.97s/it] 12%|█▏        | 30/250 [01:31<10:25,  2.84s/it] 12%|█▏        | 31/250 [01:34<10:10,  2.79s/it] 13%|█▎        | 32/250 [01:37<10:32,  2.90s/it] 13%|█▎        | 33/250 [01:40<10:55,  3.02s/it] 14%|█▎        | 34/250 [01:43<11:08,  3.10s/it] 14%|█▍        | 35/250 [01:47<11:07,  3.11s/it] 14%|█▍        | 36/250 [01:49<10:30,  2.94s/it] 15%|█▍        | 37/250 [01:52<10:32,  2.97s/it] 15%|█▌        | 38/250 [01:54<09:27,  2.68s/it] 16%|█▌        | 39/250 [01:56<08:59,  2.56s/it] 16%|█▌        | 40/250 [01:59<09:03,  2.59s/it] 16%|█▋        | 41/250 [02:02<09:53,  2.84s/it] 17%|█▋        | 42/250 [02:05<09:27,  2.73s/it] 17%|█▋        | 43/250 [02:08<09:53,  2.87s/it] 18%|█▊        | 44/250 [02:11<09:32,  2.78s/it] 18%|█▊        | 45/250 [02:14<10:06,  2.96s/it] 18%|█▊        | 46/250 [02:17<10:04,  2.96s/it] 19%|█▉        | 47/250 [02:20<10:09,  3.00s/it] 19%|█▉        | 48/250 [02:23<10:15,  3.05s/it] 20%|█▉        | 49/250 [02:26<09:51,  2.94s/it] 20%|██        | 50/250 [02:28<09:12,  2.76s/it] 20%|██        | 51/250 [02:31<08:38,  2.61s/it] 21%|██        | 52/250 [02:34<08:53,  2.69s/it] 21%|██        | 53/250 [02:36<08:56,  2.73s/it] 22%|██▏       | 54/250 [02:38<08:10,  2.50s/it] 22%|██▏       | 55/250 [02:41<07:54,  2.43s/it] 22%|██▏       | 56/250 [02:43<07:34,  2.34s/it] 23%|██▎       | 57/250 [02:45<07:56,  2.47s/it] 23%|██▎       | 58/250 [02:48<08:10,  2.56s/it] 24%|██▎       | 59/250 [02:51<08:02,  2.53s/it] 24%|██▍       | 60/250 [02:54<08:18,  2.62s/it] 24%|██▍       | 61/250 [02:56<07:51,  2.49s/it] 25%|██▍       | 62/250 [02:58<07:33,  2.41s/it] 25%|██▌       | 63/250 [03:00<07:09,  2.30s/it] 26%|██▌       | 64/250 [03:02<07:09,  2.31s/it] 26%|██▌       | 65/250 [03:05<07:20,  2.38s/it] 26%|██▋       | 66/250 [03:07<07:11,  2.35s/it] 27%|██▋       | 67/250 [03:10<08:00,  2.63s/it] 27%|██▋       | 67/250 [03:14<08:50,  2.90s/it]
ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.

Traceback (most recent call last):
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3418, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-2558020d6481>", line 11, in <module>
    agent.learn()
  File "../src/agent.py", line 138, in learn
    self.critic_1.optimizer.apply_gradients(zip(critic_1_gradient, self.critic_1.trainable_variables))
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 631, in apply_gradients
    return distribute_ctx.get_replica_context().merge_call(
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py", line 2941, in merge_call
    return self._merge_call(merge_fn, args, kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py", line 2948, in _merge_call
    return merge_fn(self._strategy, *args, **kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/autograph/impl/api.py", line 572, in wrapper
    return func(*args, **kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 682, in _distributed_apply
    update_ops.extend(distribution.extended.update(
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py", line 2494, in update
    return self._update(var, fn, args, kwargs, group)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py", line 3431, in _update
    return self._update_non_slot(var, fn, (var,) + tuple(args), kwargs, group)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py", line 3437, in _update_non_slot
    result = fn(*args, **kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/autograph/impl/api.py", line 572, in wrapper
    return func(*args, **kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 658, in apply_grad_to_update_var
    update_op = self._resource_apply_dense(grad, var, **apply_kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/adam.py", line 175, in _resource_apply_dense
    return gen_training_ops.ResourceApplyAdam(
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/util/tf_export.py", line 404, in wrapper
    return f(**kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/tensorflow/python/ops/gen_training_ops.py", line 1421, in resource_apply_adam
    _result = pywrap_tfe.TFE_Py_FastPathExecute(
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 2045, in showtraceback
    stb = value._render_traceback_()
AttributeError: 'KeyboardInterrupt' object has no attribute '_render_traceback_'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/IPython/core/ultratb.py", line 1170, in get_records
    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/IPython/core/ultratb.py", line 316, in wrapped
    return f(*args, **kwargs)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/site-packages/IPython/core/ultratb.py", line 350, in _fixed_getinnerframes
    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/inspect.py", line 1503, in getinnerframes
    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/inspect.py", line 1461, in getframeinfo
    filename = getsourcefile(frame) or getfile(frame)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/inspect.py", line 708, in getsourcefile
    if getattr(getmodule(object, filename), '__loader__', None) is not None:
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/inspect.py", line 754, in getmodule
    os.path.realpath(f)] = module.__name__
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/posixpath.py", line 391, in realpath
    path, ok = _joinrealpath(filename[:0], filename, {})
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/posixpath.py", line 424, in _joinrealpath
    newpath = join(path, name)
  File "/Users/anton/.conda/envs/reinforcement_learning/lib/python3.8/posixpath.py", line 85, in join
    elif not path or path.endswith(sep):
KeyboardInterrupt
